menas/src/main/scala/za/co/absa/enceladus/menas/repositories/VersionedMongoRepository.scala

```diff
@@ -38,13 +38,24 @@ abstract class VersionedMongoRepository[C <: VersionedModel](mongoDb: MongoDatab
 
   import scala.concurrent.ExecutionContext.Implicits.global
 
+  collection.createIndex(Indexes.compoundIndex(Indexes.text("name"), Indexes.text("description"), 
+      Indexes.text("userCreated"), Indexes.text("userUpdated"))).foreach(res => logger.info(s"Index created for $collectionName $res"))
+  
   private def getParent(oldEntity: C): MenasReference = {
     MenasReference(collection = Some(collectionBaseName), name = oldEntity.name, version = oldEntity.version)
   }
   
-  def getLatestVersions(): Future[Seq[VersionedSummary]] = {
+  def getDistinctNames(): Future[Seq[String]] = {
+    collection.distinct[String]("name").toFuture()
+  }
+
+  def getLatestVersions(searchQuery: Option[String] = None): Future[Seq[VersionedSummary]] = {
```

Comment:
Can we add some tests for this (just a few cases in `DatasetRepositoryIntegrationSuite`)?

Reply:
Cool will do